### ISSUE TITLE
No easy way to disable individual tests at 'meson test' time?

### ISSUE THREAD

**Issue by dankegel** [mesonbuild/meson#6999]
Porting a large package from CMake/CTest, I couldn't find an equivalent for ctest's --exclude-regex option in meson 0.54.

When a bug needs to be temporarily ignored on one platform without a source change, I've been using ctest's --exclude-regex flag in the script that drives ctest.

In meson, it looks like I need to implement that feature for myself.  Happily, my project
already had a wrapper script around each test (ick), so I can just throw the check in there.
But projects that don't invoke all tests through a wrapper will have a harder time.

I suppose it could be done by adding a conditional around calls to test(), but that's messy.

**Comment by jpakkane** [comment]
Typically people use [test suites](https://mesonbuild.com/Unit-tests.html#run-subsets-of-tests) for this.

**Comment by jpakkane** [comment]
We don't have this functionality ATM. It could probably be done by adding a new command line argument to `meson test`.

**Comment by ebraminio** [comment]
Nice to have, also something to exclude a suite from running.

**Comment by ebraminio** [comment]
Maybe for someone with a similar situation as me, I just learned that one can mark particular slower tests with `slow` tag like https://github.com/harfbuzz/harfbuzz/pull/2484 there is the concept of slow tests on Chromium so being achieve that with suites mechanism and using `meson test --no-suite slow` makes it so easy achieve the same.

Thank you for providing this awesome build system!

**Comment by michaelcadilhac** [comment]
Similarly, I have some tests that are very much CPU intensive, and I'd like to run them only when the user explicitly asks for them. It would be interesting to be able to disable a whole suite by default, maybe by exposing the variable `exclude_suites`.

**Comment by kloczek** [comment]
Any update about this?
It is quite important feature which lookks lioke still is missing :/

**Comment by okias** [comment]
That would be nice. We're exactly in a scenario where this would help unblock CI.

**Comment by eli-schwartz** [comment]
> Similarly, I have some tests that are very much CPU intensive, and I'd like to run them only when the user explicitly asks for them. It would be interesting to be able to disable a whole suite by default, maybe by exposing the variable `exclude_suites`.

Since Meson 0.57.0 the [`add_test_setup()`](https://mesonbuild.com/Reference-manual_functions.html#add_test_setup) function can specify an exclude_suites value for the default test setup.

u

> I couldn't find an equivalent for ctest's --exclude-regex option in meson 0.54.

But meson *did* have --no-suite, which is sort of the same thing...

> I suppose it could be done by adding a conditional around calls to test(), but that's messy.

... particularly if you're willing to add conditionals in your meson.build -- because tagging something with a suite is easier than if/endif (one line) and also allows you to configure that at test time instead of configure time.

A `meson test` option to exclude just one test without recourse to suites, (still) does not exist though.

**Comment by SilverRainZ** [comment]
I'm surprised that in 2023, meson doesn't yet have an option for packagers (not developers, packager can not edit `meson.build without patching`) to disable a certain unit test. Or maybe I misunderstood something? Can the packager achieve this function through ninja?

**Comment by xclaesse** [comment]
@SilverRainZ we are all waiting on your PR to add this move little feature :)

**Comment by xclaesse** [comment]
I just saw we already had a PR for this actually: https://github.com/mesonbuild/meson/pull/11502.

**Comment by kloczek** [comment]
> I just saw we already had a PR for this actually: #11502.

Yep and it already +half year when first time it was submitted.

**Comment by eli-schwartz** [comment]
kloczek, I'm going to make a special exception for you, and ask you not to comment if you cannot be constructive.


Edit (2026): I see I have gotten confused emoji reactions from the outside person who commented before the person I replied to, as I didn't `@mention` anyone. To be perfectly clear, the person I replied to is the person whose comment had been ***marked as abuse*** (which I guess in 2023 I had thought was obvious) and is a known toxic influence across thousands of GitHub repositories.

It takes a lot for me to be intolerant of people even interacting / commenting, but yep, kloczek manages it.

**Comment by mr-c** [comment]
Thank you @thesamesam !
